### PR TITLE
Add a reset button

### PR DESCRIPTION
**Pull request by haltman-at**
This pull request addresses issue #1283, by adding a reset command, "r", to the debugger.  There honestly isn't much more to say than that!

### CODE REVIEW

**Comment by hjaltman** [comment]
Disregard previous comment; tests seem to be working now.  No idea.

File: packages/truffle-debugger/lib/controller/sagas/index.js [hunk 1]
```diff
@@ -11,13 +11,19 @@ import * as actions from "../actions";
 
 import controller from "../selectors";
 
+import * as data from "lib/data/actions";
+import * as evm from "lib/evm/actions";
+import * as solidity from "lib/solidity/actions";
+import * as traceActions from "lib/trace/actions";
```
[the remark this answers is not in the document]
Comment (hjaltman) [reply packages/truffle-debugger/lib/controller/sagas/index.js]: Oh, I see.  I mean, that's fine -- I already was separating things out anyway.  What would be really nice, as mentioned, would be to just have a single `RESET` action, but having reducers respond to foreing actions really seemed wrong to me.  In any case yeah I can add those sagas.

File: packages/truffle-debugger/lib/evm/actions/index.js [hunk 2]
```diff
@@ -48,3 +48,8 @@ export function returnCall() {
     type: RETURN
   };
 }
+
+export const RESET_EVM = "RESET_EVM";
```
[the remark this answers is not in the document]
Comment (hjaltman) [reply packages/truffle-debugger/lib/evm/actions/index.js]: Oh, so you did.  Getting rid of the prefixing on the constants too; just including on the underlying string.

File: packages/truffle-debugger/lib/controller/sagas/index.js [hunk 3]
```diff
@@ -243,3 +247,13 @@ function *continueUntilBreakpoint () {
...
 }
+
+/**
+ * reset -- reset the state of the debugger
+ */
+function *reset() {
+  yield *data.reset();
+  yield *evm.reset();
+  yield *solidity.reset();
+  yield *trace.reset();
```
Comment (gnidan) [inline packages/truffle-debugger/lib/controller/sagas/index.js]: pretty!

**Pull request merged**